circuit-benchmarks/src/bytecode_circuit.rs

```diff
@@ -0,0 +1,142 @@
+//! State circuit benchmarks
+
+#[cfg(test)]
+mod tests {
+    use ark_std::{end_timer, start_timer};
+    use bus_mapping::evm::OpcodeId;
+    use eth_types::Field;
+    use halo2_proofs::plonk::{create_proof, keygen_pk, keygen_vk, verify_proof};
+    use halo2_proofs::poly::kzg::commitment::{KZGCommitmentScheme, ParamsKZG, ParamsVerifierKZG};
+    use halo2_proofs::poly::kzg::multiopen::{ProverSHPLONK, VerifierSHPLONK};
+    use halo2_proofs::poly::kzg::strategy::SingleStrategy;
+    use halo2_proofs::{
+        halo2curves::bn256::{Bn256, Fr, G1Affine},
+        poly::commitment::ParamsProver,
+        transcript::{
+            Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+        },
+    };
+    use rand::SeedableRng;
+    use rand_xorshift::XorShiftRng;
+    use std::env::var;
+    use zkevm_circuits::bytecode_circuit::bytecode_unroller::{unroll, UnrolledBytecode};
+    use zkevm_circuits::bytecode_circuit::dev::{get_randomness, BytecodeCircuitTester};
+
+    #[cfg_attr(not(feature = "benches"), ignore)]
+    #[test]
+    fn bench_bytecode_circuit_prover() {
+        let degree: u32 = var("DEGREE")
+            .unwrap_or_else(|_| "12".to_string())
+            .parse()
+            .expect("Cannot parse DEGREE env var as u32");
+
+        // Contract code size exceeds 24576 bytes may not be deployable on Mainnet.
+        const MAX_BYTECODE_LEN: usize = 24576;
+
+        let randomness = get_randomness();
+        let num_rows = 1 << degree;
+        const NUM_BLINDING_ROWS: usize = 7 - 1;
+        let max_bytecode_row_num = num_rows - NUM_BLINDING_ROWS;
+        let bytecode_len = std::cmp::min(MAX_BYTECODE_LEN, max_bytecode_row_num);
+        let bytecodes_num: usize = max_bytecode_row_num / bytecode_len;
+        let instance = vec![randomness; max_bytecode_row_num];
```

Comment:
I don't think this is currently required by the circuit (doesn't use instance columns), so could just be `&[&[]]` I think.

Reply:
Oh, that is because the old bytecode impl used instance role, now the latest one already changed that behavior. will fix.